argcomplete/shell_integration.py

```diff
@@ -42,7 +42,15 @@
             _ARGCOMPLETE_SHELL="zsh" \
             _ARGCOMPLETE_SUPPRESS_SPACE=1 \
             __python_argcomplete_run ${script:-${words[1]}}))
-        _describe "${words[1]}" completions -o nosort
+        local nosort=()
+        local nospace=()
+        if is-at-least 5.8; then
+            nosort=(-o nosort)
+        fi
+        if [[ "${completions-}" =~ ([^\\]): && "${match[1]}" =~ [=/:] ]]; then
```

Comment:
Note the different indexing semantics between `match` and `BASH_REMATCH`. Both versions are extracting the first match.

Reply:
The returned completion format is `completion:description`, where colons are escaped with backslashes. This looks for the character before the first unescaped colon and checks it against the same characters used for bash.